### PR TITLE
[Tests-only] acceptance tests for sharing with multiple users and groups

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

Adds acceptance test for sharing with users and groups having the same name and changing permissions and expiration date
## Related Issue

https://github.com/owncloud/core/issues/36818

## How Has This Been Tested?

- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by kiranparajuli589 — DISMISSED** [review]

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 1]
```diff
@@ -1776,6 +1776,56 @@ public function informationOfLastShareShouldInclude(
...
+		foreach ($data as $share) {
+			if ($share['file_target'] === $fileName && $share['item_type'] === $fileOrFolder) {
+				if (($share['share_type'] === SharingHelper::getShareType($userOrGroup))
+				) {
+					$share_id = $share['id'];
+				}
+			}
+		}
+
+		if ($share_id === null) {
```
Comment (jasson99) [inline tests/acceptance/features/bootstrap/Sharing.php]: Can we use assertion here?

**Review by jasson99 — COMMENTED** [review]
LGTM

**Review by kiranparajuli589 — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 2]
```diff
@@ -1776,6 +1776,54 @@ public function informationOfLastShareShouldInclude(
...
+				if (($share['share_type'] === SharingHelper::getShareType($userOrGroup))
+				) {
+					$share_id = $share['id'];
+				}
+			}
+		}
+
+		Assert::assertNotNull($share_id, "Could not find share id for " . $user);
+
+		$fd = $body->getRowsHash();
```
[the remark this answers is not in the document]
Comment (phil-davis) [reply tests/acceptance/features/bootstrap/Sharing.php]: `$fd` was used in a few other places also. A bit of history. I added a refactoring commit to rename all uses of `$fd` to $bodyRows`

**Review by phil-davis — APPROVED** [review]
I added commits to:
- do a general rename of `$fd` in all places to `$bodyRows` (I don't know what `$fd` was supposed to stand for somewhere in past history)
- rename `toggleShareActions` to `openShareActionsIfNotOpen` because that is what it does now (it never closes it)
- adjust the last test scenario that changes " sharing permissions and expiration date for both a user and a group" so that it does all the actions first, then checks the settings visible in the webUI and the settings of the shares via the API

**Pull request merged**